batch7crud-roll5/src/main/java/com/lftechnology/batch7crud/controller/EmployeeController.java

```diff
@@ -0,0 +1,169 @@
+        if (path != null) {
+            String[] pathParts = path.split("/");
+
+            if ("createProcess".equals(pathParts[1])) {
+                createProcess(request, response);
+
+            }
+
+            else if ("editProcess".equals(pathParts[2])) {
+                int id = Integer.parseInt(pathParts[1]);
+                editProcess(request, response, id);
+
+            }
+
+            else if ("deleteProcess".equals(pathParts[2])) {
+                int id = Integer.parseInt(pathParts[1]);
+                deleteProcess(request, response, id);
+            }
+
+        } else {
+            int pageNo = 1;
+            fetchData(request, response, pageNo - 1);
+        }
+
+    }
+
+    private void deleteProcess(HttpServletRequest request, HttpServletResponse response, int id) {
+        try {
+            employeeService.deleteById(id);
+
+            response.sendRedirect(employeeListUrl);
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void edit(HttpServletRequest request, HttpServletResponse response, int id) {
+        try {
+            emp = employeeService.fetchById(id);
+            request.setAttribute("employee", emp);
+            RequestDispatcher dispatcher = request.getRequestDispatcher("/WEB-INF/views/editEmployee.jsp");
+            dispatcher.forward(request, response);
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void editProcess(HttpServletRequest request, HttpServletResponse response, int id) {
+        try {
+            emp.setFirstName(request.getParameter("firstName"));
+            emp.setLastName(request.getParameter("lastName"));
+            emp.setDepartment(request.getParameter("department"));
+            emp.setAddress(request.getParameter("address"));
+
+            employeeService.edit(emp, id);
+
+            response.sendRedirect(employeeListUrl);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void create(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            RequestDispatcher dispatcher = request.getRequestDispatcher("/WEB-INF/views/createEmployee.jsp");
+            dispatcher.forward(request, response);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void createProcess(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            emp.setFirstName(request.getParameter("firstName"));
+            emp.setLastName(request.getParameter("lastName"));
+            emp.setDepartment(request.getParameter("department"));
+            emp.setAddress(request.getParameter("address"));
+
+            boolean status = employeeService.create(emp);
+
+            if (status == true) {
+                response.sendRedirect(employeeListUrl);
+            }
+
+        } catch (Exception e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)